src/ansys/pytwin/evaluate/tbrom.py

```diff
@@ -163,17 +174,70 @@ def _reduce_field_input(self, name: str, snapshot: Union[str, Path, np.ndarray])
                 self._infmcs[name][item] = mc[index]
                 index = index + 1
 
+    def project_on_mesh(
+        self, mesh: pv.DataSet, interpolate: bool, named_selection: str = None, progress_bar: bool = False
+    ):
+        """
+        Project the field ROM SVD basis onto a mesh.
+
+        Parameters
+        ----------
+        mesh: pyvista.DataSet
+            PyVista DataSet object of the targeted mesh.
+        interpolate: bool
+            Flag to indicate whether the point cloud data are interpolated (True) or not (False) on the targeted mesh.
+        named_selection: str (optional)
+            Named selection on which the mesh projection has to be performed. The default is ``None``, in which case the
+            entire domain is considered.
+        progress_bar : bool (optional)
+            Display a progress bar using ``tqdm`` when ``True``. Helpful for showing interpolation progress. Default to
+            ``False``, it is automatically set to ``True`` if ``tqdm`` is available.
+        """
+        nbmc = self.nb_modes
+        mesh_data = mesh.copy()
+        if not interpolate:  # target mesh is same as the one used to generate the ROM -> no interpolation required
+            self._outmeshbasis = self._outbasis
+            nb_data = self._outmeshbasis.shape[1]
+        else:  # interpolation required, e.g. because target mesh is different
+            if not progress_bar and _HAS_TQDM:
+                progress_bar = True
+            pointsdata = self._pointsdata.copy()
+            for i in range(0, nbmc):
+                pointsdata[str(i)] = self._outbasis[i]
+            if named_selection is not None:
+                pointsids = self.named_selection_indexes(named_selection)
+                listids = np.sort(pointsids)
+                pointsdata = pointsdata.extract_points(listids)
+            interpolated_mesh = mesh.interpolate(
+                pointsdata, sharpness=5, radius=0.0001, strategy="closest_point", progress_bar=progress_bar
```

Comment:
I guess best choice for `sharpness`, `radius` and `strategy` might be case dependent and thus it might be useful to expose them to user.

Reply:
this is advanced settings in my opinion, and this is where I would like these workflows to be tested on more use cases first

Reply:
as discussed today : we need to test more workflows and collect more feedbacks on possible limitations